src/main/java/me/jellysquid/mods/lithium/common/world/layer/LayerFactoryUtil.java

```diff
@@ -0,0 +1,20 @@
+package me.jellysquid.mods.lithium.common.world.layer;
+
+import net.minecraft.world.biome.layer.util.LayerFactory;
+import net.minecraft.world.biome.layer.util.LayerSampler;
+
+public final class LayerFactoryUtil {
+    public static <A extends LayerSampler> LayerFactory<A> memoize(LayerFactory<A> factory) {
+        return new LayerFactory<A>() {
+            final ThreadLocal<A> cached = new ThreadLocal<>();
+
+            @Override
+            public A make() {
+                if (this.cached.get() == null) {
```

Comment:
Unlikely to be a hotspot, but the repeated calls to `ThreadLocal#get` can be avoided by using a pattern such as:
```java
T val = this.cached.get();
if (val == null) {
  this.cached.set(val = computeValue());
}
return val;
```